### PR TITLE
Add system service test and update to use EE10 and MP6

### PR DESCRIPTION
**Pull request by gkwan-ibm**
- add unit tests for the system service
- address issue: https://github.com/OpenLiberty/guides-common/issues/891 and https://github.com/OpenLiberty/guides-common/issues/897

### CODE REVIEW

File: README.adoc [hunk 1]
```diff
@@ -466,6 +466,76 @@ endif::[]
...
+[source, Java, linenums, indent=0, role="code_column hide_tags=copyright"]
+----
+include::finish/system/src/test/java/io/openliberty/guides/system/SystemServiceTest.java[]
+----
+
+// file 2
+SystemService.java
+[source, Java, linenums, indent=0, role="code_column hide_tags=copyright"]
+----
+include::finish/system/src/main/java/io/openliberty/guides/system/SystemService.java[]
```
Comment (mbuamuh) [inline README.adoc:501]: The file name should be SystemServiceTest.java and not SystemTest.java; see: http://lg-dev1.fyre.ibm.com:4020/guides/grpc-intro.html#implementing-unit-tests-for-the-grpc-server-service
The test class also had compilation errors

**Review by mbuamuh — CHANGES_REQUESTED** [review]

**Review by mbuamuh — APPROVED** [review]

File: README.adoc [hunk 2]
```diff
@@ -466,6 +466,76 @@ endif::[]
 // ===============================================================================================
 
 == Testing the application
+
+You can test your application manually, but automated tests ensure code quality because they trigger a failure whenever a code change introduces a defect. In this section, you'll create unit tests for the gRPC server service and integration tests for the `query` service.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
Although you can test your application manually, automated tests ensure consistent code quality by triggering a failure whenever a code change introduces a defect. In this section, you'll create unit tests for the gRPC server service and integration tests for the `query` service.
```

File: README.adoc [hunk 3]
```diff
@@ -466,6 +466,76 @@ endif::[]
...
+=== Implementing unit tests for the gRPC server service
+
+// file 0
+system/pom.xml
+[source, xml, linenums, indent=0, role="code_column hide_tags=copyright"]
+----
+include::finish/system/pom.xml[]
+----
+
+The required dependencies are already added to the `pom.xml` Maven configuration file for you, including JUnit5, [hotspot=grpc-testing file=0]`grpc-testing`, and [hotspot=mockito-core file=0]`mockito-core` libraries. The `grpc-testing` dependency provides utilities for testing gRPC services and is used for creating a mock gRPC server for simulating client-server communication during testing. The `mockito-core` dependency is needed to enable the use of Mockito.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The `pom.xml` Maven configuration file already specifies the required dependencies, including JUnit5, [hotspot=grpc-testing file=0]`grpc-testing`, and [hotspot=mockito-core file=0]`mockito-core` libraries. The `grpc-testing` dependency provides utilities for testing gRPC services and creates a mock gRPC server that simulates client-server communication during testing. The `mockito-core` dependency enables the Mockito mocking framework.
```

File: README.adoc [hunk 4]
```diff
@@ -466,6 +466,76 @@ endif::[]
...
+// file 2
+SystemService.java
+[source, Java, linenums, indent=0, role="code_column hide_tags=copyright"]
+----
+include::finish/system/src/main/java/io/openliberty/guides/system/SystemService.java[]
+----
+
+In the `setUp()` static method, create and start the [hotspot=inProcessServer file=1]`inProcessServer` in-process gRPC server. Then, create the [hotspot=inProcessChannel file=1]`inProcessChannel` in-process channel that connects to the `inProcessServer` server running in the same JVM process. The unit tests can make calls to the gRPC server by using the same method signatures and functionalities as the gRPC client, even though they use different [hotspot=blockingStub file=1]`blockingStub` or [hotspot=asyncStub file=1]`asyncStub` stubs through the same channel.
+
+In the [hotspot=tearDown file=1]`tearDown()` static method, shutdown the `inProcessChannel` in-process channel and the `inProcessServer` in-process gRPC server.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
In the [hotspot=tearDown file=1]`tearDown()` static method, shut down the `inProcessChannel` in-process channel and the `inProcessServer` in-process gRPC server.
```

File: README.adoc [hunk 5]
```diff
@@ -466,6 +466,76 @@ endif::[]
...
+
+The [hotspot=testGetProperty file=1]`testGetProperty()` tests the unary call to retrieve a single System property value.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The [hotspot=testGetProperty file=1]`testGetProperty()` tests the unary call to retrieve a single system property value.
```

File: README.adoc [hunk 6]
```diff
@@ -466,6 +466,76 @@ endif::[]
...
+
+The [hotspot=testGetServerStreamingProperties file=1]`testGetServerStreamingProperties()` tests the server streaming call to retrieve multiple System property values with a given property prefix.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The [hotspot=testGetServerStreamingProperties file=1]`testGetServerStreamingProperties()` tests the server streaming call to retrieve multiple system property values with a given property prefix.
```

File: README.adoc [hunk 7]
```diff
@@ -466,6 +466,76 @@ endif::[]
...
+
+The [hotspot=testGetClientStreamingProperties file=1]`testGetClientStreamingProperties()` tests the client streaming call to retrieve multiple System property values with given property names.
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
The [hotspot=testGetClientStreamingProperties file=1]`testGetClientStreamingProperties()` tests the client streaming call to retrieve multiple system property values with given property names.
```

File: README.adoc [hunk 8]
```diff
@@ -466,6 +466,76 @@ endif::[]
...
+
+The [hotspot=testGetBidirectionalProperties file=1]`testGetBidirectionalProperties()` tests the bidirectional streaming call to retrieve multiple system property values with given property names.
+
+*Running the tests*
```
Comment (dmuelle) [inline README.adoc]: using bold as a substitute for heading style creates an accessibility violation. I recommend using h3 or h4 instead, depending on how the CSS handles h4

File: README.adoc [hunk 9]
```diff
@@ -488,7 +557,7 @@ The [hotspot=getUserProperties file=0]`testGetUserProperties()` tests the `/quer
 
 The [hotspot=getJavaProperties file=0]`testGetJavaProperties()` tests the `/query/properties/java` endpoint and confirms that a response is received. 
 
-=== Running the tests
+*Running the tests*
```
Comment (dmuelle) [inline README.adoc]: see comment at line 516

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review, let me know if you have any questions. Thanks

File: README.adoc (hunk 8 above)
Comment (gkwan-ibm) [reply README.adoc]: In guide (adoc), I double there is a h3 and h4 and also don't want to be a subsection of subsection.

Comment (dmuelle) [reply README.adoc]: there's an h3 heading at line 472-  

`=== Implementing unit tests for the gRPC server service`

you might try 
`==== Running the tests`

File: README.adoc [hunk 10]
```diff
@@ -466,6 +466,79 @@ endif::[]
...
+The [hotspot=testGetClientStreamingProperties file=1]`testGetClientStreamingProperties()` tests the client streaming call to retrieve multiple system property values with given property names.
+
+The [hotspot=testGetBidirectionalProperties file=1]`testGetBidirectionalProperties()` tests the bidirectional streaming call to retrieve multiple system property values with given property names.
+
+
+=== Running unit tests for the gRPC server service
+
+Because you started Open Liberty in dev mode, you can run the tests by pressing the `enter/return` key from the command-line session where you started the `system` service.
+
+If the tests pass, you see a similar output to the following example:
```
Comment (dmuelle) [inline README.adoc]:
```suggestion
If the tests pass, you see output similar to the following example:
```

File: README.adoc (hunk 8 above)
Comment (gkwan-ibm) [reply README.adoc]: addressed to use `=== Running...`

**Review by dmuelle — APPROVED** [review]
LGTM

**Comment by gkwan-ibm** [comment]
### Review changes
- [x]  pom.xml
  - java 11
  - jakarta is 10, microfile-profile is 6.0, and dependencies
  - LMP 3.7.1
- [x] server.xml
  - features version
- [x] java files
  - EPL 2.0
- [x] License file
   - EPL 2.0

### End-to-end test
- [x] update lgdev with `system-service-test` branch
- [x] do end-to-end test and review content carefully
  - clone the `system-service-test` branch
  - if index.html has changes, make sure the links work

### Cloud hosted test
- [x] generate the cloud hosted dev lab
- [x] do end-to-end test and review content
  - clone the `system-service-test` branch

**Comment by ReeceNana** [comment]
Ran guide end-to-end on `system-service-test` branch. 
No issues/errors when ran local or cloud-hosted.
Ready to merge. 

1. PR approval subject to Gilbert review
2. Re-test locally subject to Morgan

**Pull request merged**